dbms/src/Common/ConfigProcessor.cpp

```diff
@@ -311,22 +304,43 @@ XMLDocumentPtr ConfigProcessor::processConfig(const std::string & path_str)
 		for (Poco::DirectoryIterator it(merge_dir_name); it != Poco::DirectoryIterator(); ++it)
 		{
 			Poco::File & file = *it;
-			try
-			{
-				if (file.isFile() && (endsWith(file.path(), ".xml") || endsWith(file.path(), ".conf")))
-				{
-					contributing_files.push_back(file.path());
-					DocumentPtr with = dom_parser.parse(file.path());
-					merge(config, with);
-				}
-			}
-			catch (Poco::Exception & e)
+			if (file.isFile() && (endsWith(file.path(), ".xml") || endsWith(file.path(), ".conf")))
 			{
-				throw Poco::Exception("Failed to merge config with " + file.path() + ": " + e.displayText());
+				res.push_back(file.path());
 			}
 		}
 	}
 
+	return res;
+}
+
+XMLDocumentPtr ConfigProcessor::processConfig(const std::string & path_str)
+{
+	/// We need larger name pool to allow to support vast amount of users in users.xml files for ClickHouse.
+	/// Size is prime because Poco::XML::NamePool uses bad (inefficient, low quality)
+	///  hash function internally, and its size was prime by default.
+	Poco::AutoPtr<Poco::XML::NamePool> name_pool(new Poco::XML::NamePool(65521));
+	Poco::XML::DOMParser dom_parser(name_pool);
+
+	DocumentPtr config = dom_parser.parse(path_str);
+
+	std::vector<std::string> contributing_files;
+	contributing_files.push_back(path_str);
+
+	for (auto & merge_file : getConfigMergeFiles(path_str))
```

Comment:
If I am not mistaken derived type of `auto` will be `const std::string`, so the resulting type `const std::string &` is the same as in `const auto &` case.